src/latch/resources/tasks.py

```diff
@@ -627,3 +627,88 @@ def nextflow_runtime_task(cpu: int, memory: int, storage_gib: int = 50):
     ]
 
     return functools.partial(task, task_config=task_config)
+
+
+@dataclass
+class _L40sGPUInstanceSpec:
+    type: str
+    cpu: int
+    memory: int
+    gpus: int
+
+
+def l40s_gpu_task(cpu: int, memory: int, gpus: int, **kwargs):
+    """Creates a task configuration for L40s GPU instances.
+
+    Will choose the smallest instance that satisfies the resource requirements and will assign the entire instance to the task.
+
+    Args:
+        cpu: Number of vCPUs requested (4-96)
+        memory: Memory in GiB requested (32-768)
+        gpus: Number of GPUs requested (1 or 4)
+    """
+    instance_specs: list[_L40sGPUInstanceSpec] = [
+        _L40sGPUInstanceSpec(type="g6e.xlarge", cpu=4, memory=32, gpus=1),
+        _L40sGPUInstanceSpec(type="g6e.2xlarge", cpu=8, memory=64, gpus=1),
+        _L40sGPUInstanceSpec(type="g6e.4xlarge", cpu=16, memory=128, gpus=1),
+        _L40sGPUInstanceSpec(type="g6e.8xlarge", cpu=32, memory=256, gpus=1),
+        _L40sGPUInstanceSpec(type="g6e.12xlarge", cpu=48, memory=384, gpus=1),
+        _L40sGPUInstanceSpec(type="g6e.16xlarge", cpu=64, memory=512, gpus=4),
+        _L40sGPUInstanceSpec(type="g6e.24xlarge", cpu=96, memory=768, gpus=4),
+        # _L40sGPUInstanceSpec(type="g6e.48xlarge", cpu=192, memory=1536, gpus=8),
+        # # todo(aidan): add 48xlarge instance (need to debug why does not join cluster)
+    ]
+
+    selected_instance = None
+    for spec in instance_specs:
+        if (cpu <= spec.cpu and 
+            memory <= spec.memory and 
+            gpus <= spec.gpus):
+            selected_instance = spec
+            break
```

Comment:
so this is fine but it needs to be abundantly clear in the docs that we pin resources to the closest matching instance type. 

id even go so far as not making this customizable at all and having specific decorators for each instance type